lib/core/test/unit/Cardano/Wallet/CoinSelection/Internal/BalanceSpec.hs

```diff
@@ -1726,9 +1726,10 @@ prop_assetSelectionLens_givesPriorityToSingletonAssets
     :: Blind (Small (UTxOIndex TestUTxO))
     -> Property
 prop_assetSelectionLens_givesPriorityToSingletonAssets (Blind (Small u)) =
-    assetCount >= 2 ==> monadicIO $ do
+    nonAdaAssetCount >= 2 ==> monadicIO $ do
         hasSingletonAsset <- isJust <$>
-            run (UTxOIndex.selectRandom u $ WithAssetOnly asset)
+            run (UTxOIndex.selectRandom u $
+            SelectPairWith (Asset nonAdaAsset))
```

Comment:
Suggestion (after review with @sevanspowell):

Ideally, this function would test using `SelectSingleton`, rather than `SelectPair`.

For the moment, however, we need to use `SelectPair`, because the our `TokenBundle` generators produce a distribution that is **_skewed_** toward token bundles with **_non-zero_** ada quantities. This means that only a very small proportion of generated token bundles will have a single non-ada asset **_and_** no ada. Therefore, if we want to avoid adjusting these generators (which will have knock-on effects elsewhere), we can simply look for **_pairs_** that contain the asset, rather than **_singletons_**, in the knowledge that each non-ada asset is always accompanied with ada.

In a future PR, we can fix this by adjusting the distribution of token bundles.

We should:
- Add a comment here describing the motivation for using `SelectPairWith` instead of `SelectSingleton`.
- Add a TODO that reminds to change `SelectPairWith` to `SelectSingleton`, once we've moved to generating token bundles without a skewed distribution.

Reply:
Fixed in a5703949bf95343d2b2801b776e9cb5709bb8d1e.